app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+
+    for person in people:
+        person_list.append(Person(person["name"], person["age"]))
+
+    for person_dict in people:
+
+        spouse_key = "husband" if person_dict.get("husband") is not None \
+            else "wife"
+        spouse_name = person_dict.get(spouse_key)
+
+        name = person_dict["name"]
```

Comment:
Accessing `person_dict['name']` directly assumes that every dictionary in the list has a 'name' key. Consider using `get` to avoid potential `KeyError` exceptions if the 'name' key is missing.